runner/engine_test.go

```diff
@@ -185,6 +185,28 @@ func TestRerunWhenFileChanged(t *testing.T) {
 	}
 }
 
+func TestRunCommand(t *testing.T) {
+	engine, err := NewEngine("", true)
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+	err = engine.runCommand("echo Hello Air")
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+}
+
+func TestRunPreCmd(t *testing.T) {
+	engine, err := NewEngine("", true)
+	if err != nil {
+		t.Fatalf("Should not be fail: %s.", err)
+	}
+	err = engine.runPreCmd()
```

Comment:
Unit tests should not be written like this. You need to check if the corresponding values have changed

Reply:
I don't think I completely understand you!